File: app/src/main/java/sugtao4423/yandereviewer/ChromeIntent.java
package sugtao4423.yandereviewer;

import android.content.Context;
import android.graphics.Color;
import android.net.Uri;
import android.support.customtabs.CustomTabsIntent;

public class ChromeIntent{

	public ChromeIntent(Context context, Uri uri){
		CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
		builder.setShowTitle(true);
		builder.enableUrlBarHiding();
		builder.addDefaultShareMenuItem();
		builder.setToolbarColor(Color.parseColor(context.getString(R.color.primary)));
		builder.build().launchUrl(context, uri);
	}

	public ChromeIntent(Context context, String url){
		this(context, Uri.parse(url));
	}

}

File: app/src/main/java/sugtao4423/yandereviewer/ChromeIntent.kt
package sugtao4423.yandereviewer

import android.content.Context
import android.net.Uri
import androidx.browser.customtabs.CustomTabColorSchemeParams
import androidx.browser.customtabs.CustomTabsIntent
import androidx.core.content.ContextCompat

class ChromeIntent(context: Context, uri: Uri) {

    init {
        CustomTabsIntent.Builder().apply {
            setShowTitle(true)
            setUrlBarHidingEnabled(true)
            setShareState(CustomTabsIntent.SHARE_STATE_ON)
            val colorScheme = CustomTabColorSchemeParams.Builder().let {
                it.setToolbarColor(ContextCompat.getColor(context, R.color.primary))
                it.build()
            }
            setDefaultColorSchemeParams(colorScheme)
            build().launchUrl(context, uri)
        }
    }

    constructor(context: Context, url: String) : this(context, Uri.parse(url))

}